types/mongoose/index.d.ts

```diff
@@ -2626,9 +2627,9 @@ declare module "mongoose" {
       callback?: (err: any, res: T) => void): Promise<T>;
 
     /** Removes documents from the collection. */
-    remove(conditions: any, callback?: (err: any) => void): Query<void>;
-    deleteOne(conditions: any, callback?: (err: any) => void): Query<void>;
-    deleteMany(conditions: any, callback?: (err: any) => void): Query<void>;
+    remove(conditions: any | Query<any>, callback?: (err: any) => void): Query<mongodb.WriteOpResult['result']>;
+    deleteOne(conditions: any | Query<any>, callback?: (err: any) => void): Query<mongodb.WriteOpResult['result']>;
+    deleteMany(conditions: any | Query<any>, callback?: (err: any) => void): Query<mongodb.WriteOpResult['result']>;
```

Comment:
@jendrikw 
1) this PR more about response ``` Query<mongodb.WriteOpResult['result']> ```, because Query<void> don't allow use remove result
2) we use `conditions: any | Query<any>` because condition can be any object - and it's good to have such opportunity, may be we should use object instead of any

Reply:
I know it's about the return value. But you changed the parameter too, and `any | Query<any>` is not acceptable as stated in the README. Please change it to `any`.

Reply:
@jendrikw done